### PR TITLE
Changed List to add onActive

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Exposes the internal `active` state of a List via `onActive`.
Also ensures that any `onKeyDown` passed in is given to the internal Keyboard.

#### Where should the reviewer start?

I'd start by reading the unit test changes. They should give an idea on what is happening.

#### What testing has been done on this PR?

Enhanced unit tests.
Did some manual testing with the onClickItem story by adding `onActive` and `onKeyDown`

I didn't add a storybook story for this as it seemed somewhat obscure and reasonably covered by the unit tests.

#### How should this be manually tested?

Add your own `onActive` and `onKeyDown` to a story. 

#### Do Jest tests follow these best practices?

followed existing test structure

#### What are the relevant issues?

#6087 

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/List/List.js [hunk 1]
```diff
@@ -151,8 +153,15 @@ const List = React.forwardRef(
     // If onOrder is not defined but onClickItem is (e.g. the
     // List items are likely selectable), active will be the
     // index of the item which is currently active.
-    const [active, setActive] = useState();
+    const [active, setActiveState] = useState();
```
Comment (taysea) [inline src/js/components/List/List.js:156]: Should we rename `active` to `activeState` ? no change to functionality but we typically have kept these variable names tied together.

**Review by taysea — APPROVED** [review]
All looks good, soft opinion on my previous comment.

**Review by jcfilben — APPROVED** [review]
Looks good!

File: src/js/components/List/List.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/List/List.js]: Changed to leave `setActive` as was and changed new function to be `updateActive`, to make it clear it wasn't just updating a `useState`

**Pull request merged**